src/components/ModalDetailsMovie/ModalDetailsMovie.js

```diff
@@ -0,0 +1,215 @@
+    const apiKey = "879b3e71";
+    const movieID = this.props.movieID;
+    const API = `https://www.omdbapi.com/?i=${movieID}&apikey=${apiKey}`;
+
+    fetch(API)
+      .then(
+        (response) => {
+          if (response.ok) {
+            return response;
+          } else {
+            const error = new Error(
+              `Error ${response.status} ${response.statusText}. Please try agine ...`
+            );
+            error.response = response;
+            throw error;
+          }
+        },
+
+        (error) => {
+          const errMessage = new Error(
+            `${error.message}. Please check your network connection and try agine later.`
+          );
+          throw errMessage;
+        }
+      )
+      .then((response) => response.json())
+      .then((response) => {
+        if (response.Response === "True") {
+          this.setState({
+            isLoading: false,
+            movie: { ...response },
+          });
+        } else {
+          const errMessage = new Error(response.Error);
+          throw errMessage;
+        }
+      })
+      .catch((error) =>
+        this.setState({
+          isLoading: false,
+          error: error.message,
+        })
+      );
+
+    const isFavMovie = Object.values(this.context.favMovies)
+      .flat()
+      .find((movie) => movie.id === this.props.movieID);
+
+    if (!!isFavMovie)
+      this.setState({ onWatchlist: true, typeAction: "remove" });
+    else if (!isFavMovie)
+      this.setState({ onWatchlist: false, typeAction: "add" });
+  };
+
+  render() {
+    const { closeModal, movieID } = this.props;
+    const {
+      Title,
+      Year,
+      Runtime,
+      Genre,
+      Actors,
+      Plot,
+      Awards,
+      Poster,
+      imdbRating,
+      imdbVotes,
+    } = this.state.movie;
+
+    let listActors;
+    if (Actors) {
+      listActors = Actors.split(",").map((actor, index) => (
+        <li key={index}>{actor}</li>
+      ));
+    }
+
+    return (
+      <>
+        <div className={styles.fullContainer}>
+          <div className={styles.modalWrapper}>
+            {this.state.isLoading && <Loading />}
+            {this.state.error && <h3>{this.state.error}</h3>}
+
+            {!this.state.isLoading && !this.state.error && (
+              <>
+                {
+                  <SmallButtonActionWatchlist
+                    click={() =>
+                      this.context.action(
+                        this.state.typeAction,
+                        this.props.movieID,
+                        Title,
+                        Poster
+                      )
+                    }
+                    typeAction={this.state.typeAction}
+                  />
+                }
+                <button onClick={closeModal} className={styles.btnClose}>
+                  <img src={IconClose} alt="Button close details modal" />
+                </button>
+                {this.state.message && this.state.typeAction === "add" && (
```

Comment:
za dużo logiki tu trzymasz, to jest nieczytelne.
te flagi powinny być pojedyncze.
poza tym czemu nie są zdestrukturyzowane te obiekty?